main.go

```diff
@@ -51,33 +52,70 @@ func init() {
 	//+kubebuilder:scaffold:scheme
 }
 
-func main() {
+type managerOpts struct {
+	CloudConfigFile      string
+	MetricsAddr          string
+	EnableLeaderElection bool
+	ProbeAddr            string
+	WatchingNamespace    string
+	WatchFilterValue     string
+	CertDir              string
+}
 
-	// Parse args and setup logger.
-	var cloudConfigFile string
-	var metricsAddr string
-	var enableLeaderElection bool
-	var probeAddr string
-	var watchingNamespace string
-	var certDir string
-	flag.StringVar(&cloudConfigFile, "cloud-config-file", "/config/cloud-config", "Overrides the default path to the cloud-config file that contains the CloudStack credentials.")
-	flag.StringVar(&metricsAddr, "metrics-bind-address", "localhost:8080", "The address the metric endpoint binds to.")
-	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+func setFlags() *managerOpts {
+	opts := &managerOpts{}
+	flag.StringVar(
+		&opts.CloudConfigFile,
+		"cloud-config-file",
+		"/config/cloud-config",
+		"Overrides the default path to the cloud-config file that contains the CloudStack credentials.")
+	flag.StringVar(
```

Comment:
There are a few changes to input params as the upgrade desctructions required renaming and or adding and or change of default values.

Nothing to important though.